### ISSUE TITLE
Proposal: We could also include Jira as a Repository Source

### ISSUE THREAD

**Issue by slu-it** [testIT-LivingDoc/livingdoc2#64]
Most documentation (executable or otherwise) starts with an issue (e.g. change request , story etc.) within an issue management application like Atlassian's Jira.

The liefe cycle of an executable document could be as follows:

+ create a Jira issue
+ add examples to the issue (as part of the description or in special field)
+ implement the issue ATDD style:
  - implement fixture for examples in Jira issue
  - bin "executable document class" to Jira issue
  - develop functionality while continuously testing against example until it is green
+ create / amend documentation in Confluence
+ migrate examples to documentation
+ change "executable document" class binding to document in Confluence

If we want to support this, we will _simply_ have to to implement a `JiraRepository` in addition to our already planned `ConfluenceRepository`.

**Comment by benhamidene** [comment]
well as you know I do not really support this concept. But as some bdd jira plugins are successfull it seems that many teams are using Jira as a documentation source .. so why not ..

**Comment by slu-it** [comment]
I agree that noone should use Jira for any long term documentation. Issues should be forgotten as soon as they are done. But I think it would offer some benefit for the BDD movement if we could execute examples as soon as possible.
We still recommend to move to a _real_ documentation ASAP, but having the option would be nice for development. In my experience writing documentation is always the last step when implementing a user story or change request. Which also moves the implementation of acceptance tests (at least psychologically) to the end as well. But for real ATDD you should be able to implement the acceptance tests as soon as you start implementing the change.
